workers/unity/Packages/io.improbable.gdk.debug/NetStatsViewer/NetStatsTab.cs

```diff
@@ -0,0 +1,35 @@
+using Improbable.Gdk.Core.NetworkStats;
+using UnityEngine.UIElements;
+
+namespace Improbable.Gdk.Debug.NetStats
+{
+    internal abstract class NetStatsTab : VisualElement
```

Comment:
I think you can probably pull down more functionality into this base class.

---

* In `NetStatsCommandsTab` and `NetStatsWorldCommandsTab` we have a `List<(string, MessageTypeUnion)>`
* In `NetStatsUpdatesTab` we have a `List<IComponentMetaclass>` that we transform into a `string` and `MessageTypeUnion` inside `UpdateElement`. 

Instead we could have a single `protected List<(string, MessageTypeUnion)> elementInfos;` field and expose something like: 

```
protected abstract void PopulateElementInfo();
```

In this function, the implementation should fill `elementInfos` from whatever source.

---

`UpdateElement` is the same for each subclass _except_ for the `GetXxxForIndex()` call which can similarly be made abstract (`GetInfoForIndex`). Then `UpdateElement` just drops down into the base class. 

---

If you do both these changes, `InitializeTab` can be dropped into the base class:

```
public void InitializeTab() 
{
	elementInfos = PopulateElementInfo();
    
	var itemTemplate = AssetDatabase.LoadAssetAtPath<VisualTreeAsset>(UpdateRowUxmlPath);

    // Clone row for each component, fill list
    var scrollView = this.Q<ScrollView>("container");
    for (var i = 0; i < elementInfos.Count; i++)
    {
    	var element = itemTemplate.CloneTree();
        UpdateElement(i, element);
        scrollView.Add(element);
    }
}
```

---

Then one of the derived classes will just look like:

```csharp
namespace Improbable.Gdk.Debug.NetStats
{
    internal class NetStatsCommandsTab : NetStatsTab
    {
        public new class UxmlFactory : UxmlFactory<NetStatsCommandsTab>
        {
        }

        public override void PopulateElementInfo()
        {
            elementInfos = ComponentDatabase.Metaclasses
                .SelectMany(pair => pair.Value.Commands
                    .SelectMany(metaclass => new List<(string, MessageTypeUnion)>()
                    {
                        ($"{metaclass.Name}.Request",
                            MessageTypeUnion.CommandRequest(pair.Key, metaclass.CommandIndex)),
                        ($"{metaclass.Name}.Response",
                            MessageTypeUnion.CommandResponse(pair.Key, metaclass.CommandIndex))
                    })
                )
                .ToList();
        }

        // TODO: Temporary placeholder to allow sorting
        private (string name, MessageTypeUnion messageType) GetInfoForIndex(int index)
        {
            return elementInfo[index];
        }
    }
}
```

Arguably, `GetInfoForIndex` doesn't even need to be defined here, it could live in the base class for now as well.